tests/components/conditional_formatting.test.ts

```diff
@@ -363,7 +363,40 @@ describe("UI of conditional formats", () => {
       expect(dispatch).not.toHaveBeenCalledWith("ADD_CONDITIONAL_FORMAT");
       const errorString = document.querySelector(selectors.error);
       expect(errorString!.textContent).toBe("The range is invalid");
+
+      setInputValueAndTrigger(selectors.ruleEditor.range, "s!A1", "change");
+      triggerMouseEvent(selectors.buttonSave, "click");
+      await nextTick();
+      expect(dispatch).not.toHaveBeenCalledWith("ADD_CONDITIONAL_FORMAT");
+      const errorString2 = document.querySelector(selectors.error);
+      expect(errorString2!.textContent).toBe("The range is invalid");
+    });
+
+    test("display error message if and only if invalid range", async () => {
+      triggerMouseEvent(selectors.buttonAdd, "click");
+      await nextTick();
+      await nextTick();
```

Comment:
is it necessary to haev 2 nextTicks ?

Reply:
yup, we need a second tick for the selection input component to init itself. Based on its props, it dispatches a creation of selection input elements in the model then uses the new information to render the inputs created... typically something that will disappear with the stores.